File: src/_1528/go/Solution.go
package main

import "fmt"

func restoreString(s string, indices []int) string {

	var len int = len(indices)
	result := make([]byte, len)
	var value int
	for i := 0; i < len; i++ {
		value = indices[i]
		result[value] = s[i]
	}
	return string(result[:])
}
func main() {
	indices := []int{4, 5, 6, 7, 0, 2, 1, 3}
	result := restoreString("codeleet", indices)
	fmt.Println(result)
}

File: src/_1528/java/Solution.java
package _1528.java;

public class Solution {
    public String restoreString(String s, int[] indices) {
        char[] chars = s.toCharArray();
        for(int i = 0;i<indices.length;i++){
            chars[i] =(char) (chars[i] ^ chars[indices[i]]);
            chars[indices[i]] = (char)(chars[i] ^ chars[indices[i]]);
            chars[i] = (char) (chars[indices[i]] ^ chars[i]);

        }
        return new String(chars);
    }
    public static void main(String[] args) {

    }
}
